shell/platform/windows/compositor_software.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/compositor_software.h"
+
+#include "flutter/shell/platform/windows/flutter_windows_view.h"
+
+namespace flutter {
+
+CompositorSoftware::CompositorSoftware(FlutterWindowsEngine* engine)
+    : engine_(engine) {}
+
+bool CompositorSoftware::CreateBackingStore(
+    const FlutterBackingStoreConfig& config,
+    FlutterBackingStore* result) {
+  size_t size = config.size.width * config.size.height * 4;
+  void* allocation = std::calloc(size, sizeof(uint8_t));
+  if (!allocation) {
+    return false;
+  }
+
+  result->type = kFlutterBackingStoreTypeSoftware;
+  result->software.allocation = allocation;
+  result->software.height = config.size.height;
+  result->software.row_bytes = config.size.width * 4;
+  result->software.destruction_callback = nullptr;
+  result->software.user_data = nullptr;
+  return true;
+}
+
+bool CompositorSoftware::CollectBackingStore(const FlutterBackingStore* store) {
+  std::free(const_cast<void*>(store->software.allocation));
+  return true;
+}
+
+bool CompositorSoftware::Present(const FlutterLayer** layers,
+                                 size_t layers_count) {
+  FML_DCHECK(layers_count == 1);
```

Comment:
This reminds me:
This wouldn't be in this PR, but how about iterating through however many backing store layers there are and blending them per-pixel-alpha so we can composite an arbitrary number of layers? I'm doing something similar on my local branch working with platform views. If that sounds good, I can start a PR once this lands to add that here.

Reply:
That sounds reasonable but before that PR we'd first want a design doc shared and reviewed by everyone :)